libs/api/domains/financial-statements-inao/src/lib/financialStatementsInao.module.ts

```diff
@@ -1,7 +1,26 @@
-import { Module } from '@nestjs/common'
+import { DynamicModule } from '@nestjs/common'
+
+import { DataverseClient } from './client/dataverseClient'
+import {
+  DataverseClientConfig,
+  DATAVERSE_CLIENT_CONFIG,
+} from './client/dataverseClientConfig'
 import { FinancialStatementsInaoResolver } from './financialStatementsInao.resolver'
+import { FinancialStatementsInaoService } from './financialStatementsInao.service'
 
-@Module({
-  providers: [FinancialStatementsInaoResolver],
-})
-export class FinancialStatementsInaoModule {}
+export class FinancialStatementsInaoModule {
+  static register(config: DataverseClientConfig): DynamicModule {
+    return {
+      module: FinancialStatementsInaoModule,
+      providers: [
+        FinancialStatementsInaoResolver,
+        FinancialStatementsInaoService,
+        DataverseClient,
+        {
+          provide: DATAVERSE_CLIENT_CONFIG,
+          useValue: config,
+        },
+      ],
+    }
+  }
```

Comment:
We should be using our [Nest Config Module](https://docs.devland.is/libs/nest/config) instead of static register functions.

Reply:
Fixed